src/main/java/io/logz/jmx2graphite/GraphiteClient.java

```diff
@@ -85,7 +89,13 @@ public static String sanitizeMetricName(String s, boolean keepDot) {
                 sb.append(c);
             }
         }
-        return sb.toString();
+
+        try {
+            return URLEncoder.encode(sb.toString(), ENCODING);
+        } catch (UnsupportedEncodingException e) {
+            logger.error("Unsupported encoding {}.", ENCODING, e);
```

Comment:
Let's help us solve this issue by emitting the metric name we failed to encode

Reply:
I've renamed the test class, added a unit test for the sanitizeMethodName, and fixed the error message.